corehq/apps/data_dictionary/tests/test_view.py

```diff
@@ -144,3 +146,57 @@ def test_allowed_values_with_invalid_one(self):
         response_data = json.loads(response.content)
         self.assertTrue(response_data["messages"][0].startswith("Unable to save valid values longer than"))
         self._assert_allowed_values(prop, prop_payload)
+
+
+@flag_enabled('DATA_DICTIONARY')
+class DataDictionaryViewTest(TestCase):
+    domain_name = uuid.uuid4().hex
+
+    @classmethod
+    def make_web_user_with_data_dict_role(cls, email, domain, has_data_dict_access=False):
+        web_user = WebUser.create(
+            domain=domain.name,
+            username=email,
+            password="foobar",
+            created_by=None,
+            created_via=None
+        )
+        role = UserRole.create(
+            domain=domain,
+            name='Data Dictionary Access',
+            permissions=HqPermissions(view_data_dict=has_data_dict_access),
+        )
+        web_user.set_role(domain.name, role.get_qualified_id())
+        web_user.save()
+        return web_user
+
+    @classmethod
+    def setUpClass(cls):
+        super(DataDictionaryViewTest, cls).setUpClass()
+        cls.domain = create_domain(cls.domain_name)
+        cls.web_user_data_dict_access = cls.make_web_user_with_data_dict_role('has_data_dict@ex.com', cls.domain,
+                                                                            has_data_dict_access=True)
+        cls.web_user_no_data_dict_access = cls.make_web_user_with_data_dict_role('no_data_dict@ex.com', cls.domain)
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.web_user_has_data_dict_access.delete(cls.domain_name, deleted_by=None)
+        cls.web_user_no_data_dict_access.delete(cls.domain_name, deleted_by=None)
+        cls.domain.delete()
+        super(DataDictionaryViewTest, cls).tearDownClass()
+
+    def setUp(self):
+        self.url = reverse('data_dictionary', args=[self.domain_name])
+        self.client = Client()
```

Comment:
out of curiosity, why are these in `setUp` rather than `setUpClass`?  Does the client need to be reinitialized or something?

Reply:
Originally I thought client needs to be reinitialized to login as different users but doesn't seem to be the case. Moved it to setUpClass in [b366970](https://github.com/dimagi/commcare-hq/pull/32345/commits/b36697058214b20564c6519b56405d6d7d302bb5)